modules/rostests/apitests/user32/PrivateExtractIcons.c

```diff
@@ -3,38 +3,132 @@
  * LICENSE:         LGPLv2.1+ - See COPYING.LIB in the top level directory
  * PURPOSE:         Test for PrivateExtractIcons
  * PROGRAMMER:      Hermes Belusca-Maito
+ *                  Doug Lyons <douglyons@douglyons.com>
  */
 
 #include "precomp.h"
+#include <stdio.h>
+
+BOOL FileExists(LPCWSTR FileName)
+{
+    DWORD Attribute = GetFileAttributesW(FileName);
+
+    return (Attribute != INVALID_FILE_ATTRIBUTES &&
+            !(Attribute & FILE_ATTRIBUTE_DIRECTORY));
+}
+
+BOOL ResourceToFile(INT i, LPCWSTR FileName)
+{
+    FILE *fout;
+    HGLOBAL hData;
+    HRSRC hRes;
+    LPVOID lpResLock;
```

Comment:
```suggestion
    PVOID lpResLock;
```
could be renamed to pResLock as well.